src/core/internal/convert.d

```diff
@@ -624,7 +624,14 @@ template floatFormat(T) if (is(T:real) || is(T:ireal))
     static if (T.mant_dig == 24)
         enum floatFormat = FloatFormat.Float;
     else static if (T.mant_dig == 53)
-        enum floatFormat = FloatFormat.Double;
+    {
+        // Double precision, or real == double
+        static if (T.sizeof == double.sizeof)
+            enum floatFormat = FloatFormat.Double;
+        // 80-bit real with rounding precision set to 53 bits.
+        else static if (T.sizeof == real.sizeof)
```

Comment:
I wonder how much code I've written that assumes that anything with 53 bits of mantissa has the size of double or cdouble.

Reply:
That's fine for LDC. - How do you select this format with GDC, Iain?

Reply:
Outside of phobos?  I doubt it's very much widespread.

Reply:
I meant compiler cmdline flags or such to enable x87 real with `mant_dig == 53`.

Reply:
@kinke - sorry! I was responding to the first comment. :-)

`mant_dig == 53` is the default for i586-freebsd targets, there's no way to override this as far as I know ([see freebsd support headers](https://github.com/D-Programming-GDC/gcc/blob/cfc2ade5d0f5dc255b49d496c0d6d845e0cddcce/gcc/config/i386/freebsd.h#L120-L123)).

Reply:
For C and C++, okay, but it should be trivial to use the regular 64 for GDC, given that you're setting it to 64 right here in `_d_run_main2` for all programs with D main.

Reply:
@kinke - it's hardcoded in the target headers, so there'll have to be some messing around and convincing maintainers that it's needed.

By and large, I don't think it's really worth caring about, supporting it is very much minimal, and we mostly only care about unittests.

And... I've just ran std.math unittests with working fpu setting, and tests are failing in more places than before.  Possibly because real literals are truncated to 53-bits by gcc backend.  So I'll just remove the FPU setting as it's an unsupported configuration.